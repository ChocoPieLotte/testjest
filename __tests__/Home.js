import 'react-native';
import React from 'react';
import Home from '../src/scenes/Home';

import {shallow} from 'enzyme';

describe('Home', () => {
  const navigation = {
    navigate: jest.fn()
  };
  const wrapper = shallow(<Home navigation={navigation} />);

  it('renders correctly', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should call navigate', () => {
    const button = wrapper.find('TouchableOpacity');
    button.simulate('press');
    expect(navigation.navigate).toBeCalledWith('Counter');
  });
});
