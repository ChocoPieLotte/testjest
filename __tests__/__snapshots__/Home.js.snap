// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#3498db",
          "borderRadius": 6,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontWeight": "bold",
            "margin": 12,
          }
        }
      >
        Redux Counter
      </Text>
    </TouchableOpacity>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#3498db",
            "borderRadius": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontWeight": "bold",
              "margin": 12,
            }
          }
        >
          Redux Counter
        </Text>
      </TouchableOpacity>,
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontWeight": "bold",
              "margin": 12,
            }
          }
        >
          Redux Counter
        </Text>,
        "onPress": [Function],
        "style": Object {
          "backgroundColor": "#3498db",
          "borderRadius": 6,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Redux Counter",
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "black",
            "fontWeight": "bold",
            "margin": 12,
          },
        },
        "ref": null,
        "rendered": "Redux Counter",
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "backgroundColor": "#3498db",
              "borderRadius": 6,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontWeight": "bold",
                "margin": 12,
              }
            }
          >
            Redux Counter
          </Text>
        </TouchableOpacity>,
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "black",
                "fontWeight": "bold",
                "margin": 12,
              }
            }
          >
            Redux Counter
          </Text>,
          "onPress": [Function],
          "style": Object {
            "backgroundColor": "#3498db",
            "borderRadius": 6,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Redux Counter",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "black",
              "fontWeight": "bold",
              "margin": 12,
            },
          },
          "ref": null,
          "rendered": "Redux Counter",
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
